### ISSUE TITLE
<algorithm>: std::ranges::unique returns the wrong value for already unique input

### ISSUE THREAD

**Issue by bkentel** [microsoft/STL#1560]
```
std::vector<int> values {1, 2, 3};
auto const [first, last] = std::ranges::unique(values);
values.erase(first, last);
// oops now values is {1, 2}
```

**Comment by miscco** [comment]
Thanks a lot for findings this bug. 

I verified that we indeed do not return the correct result in case of an unperturbed range and provided a fix.

**Comment by silverqx** [comment]
This bug was merged on 7. Jan, yesterday came out  Visual Studio 16.9 and this fix is still not in, when this problem will be released? @StephanTLavavej @miscco Thx for a response.

It looks that it can fix also the following problem [std::ranges::unique returns incorrect result on one item container](https://developercommunity.visualstudio.com/t/stdrangesunique-returns-incorrect-result-on-one-it/1359655).

**Comment by StephanTLavavej** [comment]
As documented in https://github.com/microsoft/STL/wiki/Changelog (search for #1561), this fix shipped in VS 2019 16.10 Preview 1. While this may seem surprising, it's because VS releases are *deeply* pipelined. In addition to the latency between merging a change (simultaneously on GitHub and our MSVC-internal mirror repo) and shipping in a public VS release, the release cycle also "locks down" near the end - for example, arbitrary changes couldn't automatically ship in 16.9 Preview 4 and 5, they had to be manually approved and ported (it's intentionally a fair amount of process/work to ensure that only critical, non-destabilizing fixes are ported before the final production release). In this case, we didn't manually port the fix.

**Comment by silverqx** [comment]
Thank you for your clarification, this release process is absolutely ok and the purpose is to minimize unstable code in releases, it is understandable.
#1561 will be shipped in _VS 2019 16.10 Preview 1_, thank you for your reply.
